### PR TITLE
Added support for shards total count in Elasticsearch Module

### PR DESCRIPTION
**Pull request by philippkahr**
Hi,

I just added the support for the `shards_stats.total_count` which can be nice for alerting, since a lot of shards on a single node can impact performance.

Question: if I fix it here, that means it is fixed for the Elastic Agent integration package as well?

```
GET /_nodes/_all/stats?filter_path=*.*.*.shard_stats.*
```

Retrieves:

```
{
  "nodes": {
    "NXSI_2XFQaC2HH7ZvdTJuQ": {
      "indices": {
        "shard_stats": {
          "total_count": 0
        }
      }
    },
    "36aFMv40QX6MBZ4u1DsCCQ": {
      "indices": {
        "shard_stats": {
          "total_count": 0
        }
      }
    },
    "g2N_KasgRkuBEvD529XfhQ": {
      "indices": {
        "shard_stats": {
          "total_count": 1753
        }
      }
    },
    "ObCmi3u1Rhiyp4JTosFXdw": {
      "indices": {
        "shard_stats": {
          "total_count": 866
        }
      }
    },
    "AW982puiTbq-RNJpg23QEw": {
      "indices": {
        "shard_stats": {
          "total_count": 866
        }
      }
    }
  }
}
```

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Review by miltonhultgren — APPROVED** [review]
These changes look fine but we'll need to update the mappings for the xpack more which live [here](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/template-resources/src/main/resources/monitoring-es-mb.json) in the Elasticsearch repo as well as the mappings for the Integration which live [here](https://github.com/elastic/integrations/blob/main/packages/elasticsearch/data_stream/node_stats/fields/fields.yml).

File: metricbeat/module/elasticsearch/node_stats/data.go [hunk 1]
```diff
@@ -112,6 +112,9 @@ var (
 					"count": c.Int("query_total"),
 				},
 			}),
+			"shard_stats": s.Object{
+				"total_count": c.Int("total_count"),
+			},
```
Comment (miltonhultgren) [inline metricbeat/module/elasticsearch/node_stats/data.go:117]:
```suggestion
			"shard_stats": c.Dict("shard_stats", s.Schema{
				"total_count": c.Int("total_count"),
			}, c.DictOptional),
```

**Review by miltonhultgren — CHANGES_REQUESTED** [review]
I should have reviewed this more carefully yesterday! 

We first need to wrap the access in a `c.Dict` to go into `indices.shard_stats` in the response, where we'll then find the field `total_count`. We also make that `Dict` optional so in case it's not there (as in older versions) we just skip grabbing that stat.

This should make the tests pass.

File: metricbeat/module/elasticsearch/node_stats/data.go (hunk 1 above)
Comment (philippkahr) [reply metricbeat/module/elasticsearch/node_stats/data.go:117]: Ah that makes sense!

**Review by miltonhultgren — APPROVED** [review]

**Pull request merged**